rust/core-lib/src/plugins/manager.rs

```diff
@@ -145,6 +146,35 @@ impl PluginManager {
         }
     }
 
+    // NOTE: This is a temporary API just for tracing.
+    fn request_trace_rpc_sync(&self, method: &str, params: &Value)
+        -> Vec<Result<Value, xi_rpc::Error>>
+    {
+        let num_global_plugins = self.global_plugins.len();
+        let num_non_global_plugins = self.buffer_plugins.len();
```

Comment:
`self.buffer_plugins` isn't a list of plugins, but a list of buffers which _have_ plugins, so this won't be returning the correct value if any buffers have more than one plugin running.

Your logic below where you actually dispatch to all plugins is correct, though.

Reply:
Yeah, you're right.  I might just remove the with_capacity then.  There's no easy way a priori of iterating the data structure to know it.